internal/telemetry/telemetry.go

```diff
@@ -7,17 +7,50 @@
 // Datadog regarding usage of an APM library such as tracing or profiling.
 package telemetry
 
-// ProductChange enqueues an app-product-change event that signals a product has been turned on/off.
+import "gopkg.in/DataDog/dd-trace-go.v1/internal/appsec"
+
+// ProductStart signals that the tracer or profiler has started with
+// some configuration information. The telemetry event emitted depends
+// on which product is starting, as well as whether an app-started event
+// has already been sent.
+func (c *client) ProductStart(namespace Namespace, configuration []Configuration) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if c.started {
+		switch namespace {
+		case NamespaceProfilers:
+			c.productChange(NamespaceProfilers, true, configuration)
+		case NamespaceTracers:
+			// Since appsec is integrated with the tracer, we sent an app-product-change
+			// update about appsec when the tracer starts. Any tracer-related configuration
+			// information can be passed along here as well.
+			c.productChange(NamespaceASM, appsec.Enabled(), configuration)
```

Comment:
This is super nitpicky but can we add a default case here that logs a debug message that an unidentified namespace was provided?